lib/telemetry_influx_db.ex

```diff
@@ -133,20 +147,31 @@ defmodule TelemetryInfluxDB do
     Supervisor.stop(pid)
   end
 
-  defp child_specs(:http, config), do: http_child_specs(config)
-  defp child_specs(:udp, config), do: udp_child_specs(config)
+  defp child_specs(protocol, config) do
+    publisher_child_specs(protocol, config) ++ common_child_specs(config)
+  end
+
+  defp publisher_child_specs(:http, config), do: [HTTP.Pool.child_spec(config)]
+
+  defp publisher_child_specs(:udp, config),
+    do: [%{id: UDP.Connector, start: {UDP.Connector, :start_link, [config]}}]
 
-  defp http_child_specs(config) do
+  defp common_child_specs(config) do
     [
-      HTTP.Pool.child_spec(config),
-      %{id: HTTP.Handler, start: {HTTP.EventHandler, :start_link, [config]}}
+      %{id: EventHandler, start: {EventHandler, :start_link, [config]}},
+      %{id: BatchReporter, start: {BatchReporter, :start_link, [batch_reporter_options(config)]}}
     ]
   end
 
-  defp udp_child_specs(config) do
+  def batch_reporter_options(config) do
     [
-      %{id: UDP.Connector, start: {UDP.Connector, :start_link, [config]}},
-      %{id: UDP.Handler, start: {UDP.EventHandler, :start_link, [config]}}
+      name: BatchReporter.get_name(config),
+      batch_size: config.batch_size,
+      report_fn: fn events, config ->
+        events
+        |> Enum.join("\n")
+        |> (&config.publisher.publish(&1, config)).()
+      end
```

Comment:
Given the approach discussed above, this function would take a list of `{event, config}` tuples, grab all of the events and join them, and grab either the first or last `config` and pass that to `publish`.